lib/ret_web/controllers/api/v1/hub_controller.ex

```diff
@@ -1,16 +1,101 @@
 defmodule RetWeb.Api.V1.HubController do
   use RetWeb, :controller
+  import RetWeb.ApiHelpers
 
   alias Ret.{Hub, Scene, SceneListing, Repo}
 
   import Canada, only: [can?: 2]
+  import Ecto.Query, only: [where: 3, preload: 2]
 
   # Limit to 1 TPS
   plug(RetWeb.Plugs.RateLimit)
 
   # Only allow access to remove hubs with secret header
   plug(RetWeb.Plugs.HeaderAuthorization when action in [:delete])
 
+  @show_request_schema %{
+                         "type" => "object",
+                         "properties" => %{
+                           "created_by_account_email" => %{
+                             "type" => "string",
+                             "format" => "email"
+                           },
+                           "hub_sids" => %{
+                             "type" => "array",
+                             "items" => %{
+                               "type" => "string"
+                             }
+                           }
+                         }
+                       }
+                       |> ExJsonSchema.Schema.resolve()
+
+  def show(conn, params) do
+    exec_api_show(conn, params, @show_request_schema, &render_hub_records/2)
+  end
+
+  defp filter_by_created_by_account(query, %Ret.Account{} = account) do
+    query
+    |> preload(:created_by_account)
+    |> where([hub], hub.created_by_account_id == ^account.account_id)
+  end
+
+  defp filter_by_created_by_account(query, _params) do
+    query
+  end
+
+  defp filter_by_created_by_account_email(query, %{"created_by_account_email" => email}) do
+    created_by_account = Ret.Account.account_for_email(email)
+    query |> filter_by_created_by_account(created_by_account)
+  end
+
+  defp filter_by_created_by_account_email(query, _params) do
+    query
+  end
+
+  defp filter_by_hub_sids(query, %{"hub_sids" => hub_sids}) do
+    query |> filter_by_hub_sids(hub_sids)
+  end
+
+  defp filter_by_hub_sids(query, hub_sids) when is_list(hub_sids) do
+    query |> where([hub], hub.hub_sid in ^hub_sids)
+  end
+
+  defp filter_by_hub_sids(query, _params) do
+    query
+  end
+
+  defp filter_by_can_join(hubs, %Ret.Account{} = account) do
+    hubs
+    |> Ret.Repo.preload(:hub_bindings)
+    |> Enum.filter(fn hub ->
+      account |> can?(join_hub(hub))
+    end)
+  end
+
+  defp filter_by_can_join(_hubs, _account) do
+    []
+  end
+
+  defp render_hub_records(conn, params) do
+    account = Guardian.Plug.current_resource(conn)
+
+    hubs =
+      Ret.Hub
+      |> filter_by_created_by_account_email(params)
+      |> filter_by_hub_sids(params)
+      |> Ret.Repo.all()
+      |> filter_by_can_join(account)
+
+    views =
+      hubs
+      |> Enum.map(fn hub ->
+        Phoenix.View.render(RetWeb.Api.V1.HubView, "show.json", %{hub: hub}) |> Map.get(:hubs) |> hd
```

Comment:
I think we should do a new view for this route and map `hub` to `room`. As we make more APIs public we can continue to make this change.